### PR TITLE
Revise warning message for triangular windows used with AirFlowNetwork

### PR DESCRIPTION
**Pull request by lgu1234**
Addresses #5384 
Revise a warning message with better description and added a section to explain how effective width and height are calculated in IO Ref

### CODE REVIEW

File: doc/src/docs/InputOutputReference/01c-InputOutputReference.md [hunk 1]
```diff
@@ -2439,6 +2439,12 @@ The AirflowNetwork:Multizone:Surface object allows a heat transfer surface or su
 
 An interior heat transfer surface (BuildingSurface:Detailed) whose surface name is used as the input for the Outside Boundary Condition Object field represents a floor without ground contact and is not allowed as an AirflowNetwork:Multizone:Surface. A heat transfer surface defined in the BuildingSurface:Detailed:ExteriorNaturalVentedCavity is also not allowed.
 
+When a triangular subsurface is used, the model provides a warning and treats this subsurface as rectangular. The effective width and height calculated in the ProcessSurfaceVertices function of the SurfaceGeometry module are used to represent a rectangular subsurface.
+
+Effective width = 0.75 * Length between Vertex 1 and Vertex 2
+
+Effective width = 4 * Area / ( 3 * Length between Vertex 2 and Vertex 3 )
```
Comment (mjwitte) [inline doc/src/docs/InputOutputReference/01c-InputOutputReference.md:2446]: @lgu1234 These both say "width".  One should be "height"?  Also, will need a unit test.

**Comment by lgu1234** [comment]
@mjwitte You are right. One of them is height. I am going to make changes. I though of a unit test. The change I made is a warning message. How can I perform the unit test for messages. Please advise.

**Comment by mjwitte** [comment]
There are several output checking functions in energyplusfixture.cc  For this case, you can check if there is any error message with `EXPECT_FALSE(has_err_output()); or EXPECT_TRUE . . .` examples are [here](https://github.com/NREL/EnergyPlus/pull/5422/files#diff-0d9f8c682689a9d64ff33dcb1c7443f8R147)

or you can compare the full content of an error message with `compare_err_stream( error_string, reset_stream_flag)`
example [here](https://github.com/NREL/EnergyPlus/blob/da43757921bbf1606cd38230b5972b4b8396d692/tst/EnergyPlus/unit/BranchNodeConnections.unit.cc#L111-L119)
It can be a bit messy to build the full error message string, so one shortcut is to let the expect fail, then copy the full error message from the console output and paste it into the delimited string code and clean that up.

**Comment by lgu1234** [comment]
@mjwitte Corrected a typo in IO Ref and added a unit test for a warning message. The branch was updated a while ago.

**Comment by mjwitte** [comment]
@lgu1234 Not sure what happened with the i386-Windows tests.  Since this branch is a bit behind, please merge in develop and will check the CI results again.

**Comment by lgu1234** [comment]
@mjwitte I am going to merge in develop and hope no test failure in i386-Windows.

File: tst/EnergyPlus/unit/AirflowNetworkBalanceManager.unit.cc [hunk 2]
```diff
@@ -216,4 +216,127 @@ namespace EnergyPlus {
 		SurfaceWindow.deallocate();
 
 	}
+
+	TEST_F( EnergyPlusFixture, AirflowNetworkBalanceManager_TestWarningMessage ) {
```
Comment (mjwitte) [inline tst/EnergyPlus/unit/AirflowNetworkBalanceManager.unit.cc]: @lgu1234 I would make the test name more specific, perhaps "AirflowNetworkBalanceManager_TestTriangulerWindowWarning"

File: tst/EnergyPlus/unit/AirflowNetworkBalanceManager.unit.cc [hunk 3]
```diff
@@ -216,4 +216,127 @@ namespace EnergyPlus {
...
+		AirflowNetworkNodeData.deallocate( );
+		AirflowNetworkCompData.deallocate( );
+		MultizoneExternalNodeData.deallocate( );
+		Zone.deallocate( );
+		Surface.deallocate( );
+		SurfaceWindow.deallocate( );
+
+	}
+
 }
```
Comment (mjwitte) [inline tst/EnergyPlus/unit/AirflowNetworkBalanceManager.unit.cc]: @lgu1234 This file needs a return at the end of it.

**Comment by mjwitte** [comment]
Tested with the defect file and the new warning is good.  Unit test runs.  Will merge after minor unit test comment is addressed.

File: doc/src/docs/InputOutputReference/01c-InputOutputReference.md [hunk 4]
```diff
@@ -2439,6 +2439,12 @@ The AirflowNetwork:Multizone:Surface object allows a heat transfer surface or su
 
 An interior heat transfer surface (BuildingSurface:Detailed) whose surface name is used as the input for the Outside Boundary Condition Object field represents a floor without ground contact and is not allowed as an AirflowNetwork:Multizone:Surface. A heat transfer surface defined in the BuildingSurface:Detailed:ExteriorNaturalVentedCavity is also not allowed.
 
+When a triangular subsurface is used, the model provides a warning and treats this subsurface as rectangular. The effective width and height calculated in the ProcessSurfaceVertices function of the SurfaceGeometry module are used to represent a rectangular subsurface.
+
+Effective width = 0.75 * Length between Vertex 1 and Vertex 2
+
+Effective height = 4 * Area / ( 3 * Length between Vertex 2 and Vertex 3 )
```
Comment (mjwitte) [inline doc/src/docs/InputOutputReference/01c-InputOutputReference.md:2446]: @lgu1234 Sorry - also need to escape the asterisks here.  They are being interpreted as italics.

Comment (lgu1234) [inline doc/src/docs/InputOutputReference/01c-InputOutputReference.md:2446]: @mjwitte I am going to use "x" to replace "*".

File: tst/EnergyPlus/unit/AirflowNetworkBalanceManager.unit.cc (hunk 2 above)
Comment (lgu1234) [inline tst/EnergyPlus/unit/AirflowNetworkBalanceManager.unit.cc]: @mjwitte OK. I am going to make a change.

File: doc/src/docs/InputOutputReference/01c-InputOutputReference.md (hunk 4 above)
Comment (mjwitte) [inline doc/src/docs/InputOutputReference/01c-InputOutputReference.md:2446]: Ok, or you can use \ in front of the *

File: tst/EnergyPlus/unit/AirflowNetworkBalanceManager.unit.cc (hunk 3 above)
Comment (lgu1234) [inline tst/EnergyPlus/unit/AirflowNetworkBalanceManager.unit.cc]: @mjwitte I am going to ass a return.

**Comment by lgu1234** [comment]
@mjwitte I just upload the revisions and merge in develop.

**Pull request merged**